src/circuit_proof/r1cs.rs

```diff
@@ -1,26 +1,38 @@
 #![allow(non_snake_case)]
```

Comment:
Light bikeshed: how about a friendlier name than `r1cs`? I just remember how it's intimidating when very specialized terms are used in things I'm trying to learn, making it harder for now good reason. E.g. it's not like we have rank-0 and rank-2 constraint systems too, so we need to distinguish between them.

How about `constraints.rs`?

Reply:
@hdevalence yeah, good point. My original comment relates to reviewing the code as a whole, so I think it still applies regardless of whether this module is exposed to the users.

Reply:
@oleganza that makes sense. I just think that `constraints` is too generic... whereas `constraint_system` is a better description of what the code is creating. /shrug both are better than r1cs though

Reply:
Just my 2c, but I think `r1cs` is fine, it's short, unique, and a specific term of art used by other systems (not just something we made up).

Reply:
Alright let's keep `r1cs` for now :)